packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -283,8 +283,8 @@ class CupertinoTextField extends StatefulWidget {
     this.cursorRadius = const Radius.circular(2.0),
     this.cursorOpacityAnimates = true,
     this.cursorColor,
-    this.selectionHeightStyle = ui.BoxHeightStyle.tight,
```

Comment:
I see this a lot in the framework where higher level widgets set their default to the same as their basic counterpart, i.e. TextField sets the same defaults as EditableText. Is this a pattern we want to be strict about? Can we just keep the defaults in EditableText and have TextField fall back on that. This PR does this for selectionHeightStyle and selectionWidthStyle, but i'm curious what others think.

Reply:
This has come up a few times and I have softly decided to support setting the default everywhere. I just sent you some notes I wrote about it.

Reply:
Hm it seems a drawback of omitting the default value here is that the property becomes nullable? It seems like a nice usability win to have a non-nullable property.

Also, not setting the default property also makes this particular change a breaking change. I'd lean towards avoid a breaking change if possible.

Reply:
Ah but since `EditableText.defaultSelectionHeightStyle` and `EditableText.defaultSelectionWidthStyle` are non-const, we can't actually use them as default values in a `const` constructor? If that's the case, I'm OK with changing these to be null by default.

Reply:
Yeah it won't be possible to set the defaults because the selectionStyle getters are non-const. Is this considered a breaking change because `TextField().selectionHeightStyle/selectionWidthStyle` now return `null`?

Reply:
> Is this considered a breaking change because `TextField().selectionHeightStyle/selectionWidthStyle` now return `null`?

Strictly speaking, yes.

One way to avoid this breaking change would be by making `TextField().selectionHeightStyle/selectionWidthStyle` getters that return `EditableText.defaultSelectionWidthStyle/defaultSelectionWidthStyle` instead of `null`. That feels a bit weird though, I suspect the breaking change is the better solution long-term.